src/Spatial/Euclidean/LIneSegment2D.cs

```diff
@@ -0,0 +1,224 @@
+        /// <param name="endPoint">the ending point of the line segment</param>
+        public LineSegment2D(Point2D startPoint, Point2D endPoint)
+        {
+            if (startPoint == endPoint)
+            {
+                throw new ArgumentException("The Line2D starting and ending points cannot be identical");
+            }
+
+            this.StartPoint = startPoint;
+            this.EndPoint = endPoint;
+        }
+
+        /// <summary>
+        /// Gets the distance from <see cref="StartPoint"/> to <see cref="EndPoint"/>
+        /// </summary>
+        [Pure]
+        public double Length => this.StartPoint.DistanceTo(this.EndPoint);
+
+        /// <summary>
+        /// Gets a normalized vector in the direction from <see cref="StartPoint"/> to <see cref="EndPoint"/>
+        /// </summary>
+        [Pure]
+        public Vector2D Direction => this.StartPoint.VectorTo(this.EndPoint).Normalize();
+
+        /// <summary>
+        /// Returns a value that indicates whether each pair of elements in two specified lines is equal.
+        /// </summary>
+        /// <param name="left">The first line to compare</param>
+        /// <param name="right">The second line to compare</param>
+        /// <returns>True if the lines are the same; otherwise false.</returns>
+        public static bool operator ==(LineSegment2D left, LineSegment2D right)
+        {
+            return left.Equals(right);
+        }
+
+        /// <summary>
+        /// Returns a value that indicates whether any pair of elements in two specified lines is not equal.
+        /// </summary>
+        /// <param name="left">The first line to compare</param>
+        /// <param name="right">The second line to compare</param>
+        /// <returns>True if the lines are different; otherwise false.</returns>
+        public static bool operator !=(LineSegment2D left, LineSegment2D right)
+        {
+            return !left.Equals(right);
+        }
+
+        /// <summary>
+        /// Returns a new <see cref="Line2D"/> from a pair of strings which represent points.
+        /// See <see cref="Point2D.Parse(string, IFormatProvider)" /> for details on acceptable formats.
+        /// </summary>
+        /// <param name="startPointString">The string representation of the first point.</param>
+        /// <param name="endPointString">The string representation of the second point.</param>
+        /// <returns>A line segment from the first point to the second point.</returns>
+        public static LineSegment2D Parse(string startPointString, string endPointString)
+        {
+            return new LineSegment2D(Point2D.Parse(startPointString), Point2D.Parse(endPointString));
+        }
+
+        /// <summary>
+        /// Translates a line according to a provided vector
+        /// </summary>
+        /// <param name="vector">A vector to apply</param>
+        /// <returns>A new translated linesegment</returns>
+        public LineSegment2D Translate(Vector2D vector)
```

Comment:
We can use TranslateBy to maximize sameness, think it is named with suffix By in other parts of the lib.

Reply:
we said that before...if only I could remember ... changing it :)